app/page/info.tsx

```diff
@@ -5,121 +5,113 @@ import { Link } from "wouter"
 import { NavbarLayout } from "~app/layout/navbar"
 import { Path } from "~app/path"
-import { useAccount } from "~app/storage"
-import { format } from "~packages/util/json"
+import { useAccount, useUserOperationLogs } from "~app/storage"
+import {
+  UserOperationStatus,
+  type UserOperationLog
+} from "~background/storage/local"
+import { UserOperation } from "~packages/bundler"
+import address from "~packages/util/address"
 import type { HexString } from "~typing"
 
+type UserOperationUi = {
+  status: UserOperationStatus
+  hash: HexString
+}
+
 export function Info() {
+  const explorerUrl = "https://jiffyscan.xyz/"
+
   const { provider } = useProviderContext()
+  const userOpLogs = useUserOperationLogs()
   const account = useAccount()
+  const [chainName, setChainName] = useState<string>("")
   const [balance, setBalance] = useState<bigint>(0n)
-  const [transactionHashes, setTransactionHashes] = useState<HexString[]>([""])
-  const [internalTransactionHashes, setInternalTransactionHashes] = useState<
-    HexString[]
-  >([""])
-  const [explorerUrl, setExplorerUrl] = useState<string>("")
+  const [balanceLoading, setBalanceLoading] = useState<boolean>(false)
+  const [userOpsUi, setUserOpsUi] = useState<UserOperationUi[]>([])
 
   useEffect(() => {
+    // Retrieve chainName when the Info page is opened
     const asyncFn = async () => {
-      const explorer = getExplorerUrl((await provider.getNetwork()).name)
-      setExplorerUrl(explorer)
+      const chain = getChainName((await provider.getNetwork()).name)
+      setChainName(chain)
+    }
 
+    asyncFn()
+  }, [])
+
+  useEffect(() => {
+    setBalanceLoading(true)
+    // Periodically check the balance of the account
+    const asyncIntervalFn = async () => {
       const balance = await provider.getBalance(account.address)
+      setBalanceLoading(false)
       setBalance(balance)
-
-      const txHashes = await accountTransactions(explorer, account.address)
-      setTransactionHashes(txHashes)
-
-      const internalTxHashes = await accountInternalTransactions(
-        explorer,
-        account.address
-      )
-      setInternalTransactionHashes(internalTxHashes)
     }
+    const id = setInterval(() => {
+      asyncIntervalFn().catch((e) => console.log(e))
+    }, 3333) // Every 3.333 seconds
 
-    asyncFn()
+    return () => {
+      clearInterval(id)
+    }
   }, [])
 
+  useEffect(() => {
+    // Update userOpsUiData when userOpLogs change
+    const userOpsUiData = getUserOpsUiData(
+      userOpLogs,
+      account.chainId,
+      account.address
+    )
+    setUserOpsUi(userOpsUiData)
+  }, [userOpLogs, account])
+
   return (
     <NavbarLayout>
       {account.address && (
-        <AccountAddress account={account.address} explorerUrl={explorerUrl} />
-      )}
-      {balance && <AccountBalance balance={balance} />}
-      <SwitchToSendPage />
-      {transactionHashes && (
-        <AccountTransactions
+        <AccountAddress
```

Comment:
I saw these sub components like `AccountAddress` and `AccountBalance` don't have complex sturcture. I would suggest to inline them.